### PR TITLE
Feature/222 assign to me switch

### PR DESCRIPTION
**Pull request by timcowlishaw**
## Changes in this PR:
This fulfils two user needs - one the need to see at a glance who is working on a judgment, and two the need to take responsibility for a new judgment as it comes in.

After some discussion, we simplified the design from the azure template to better tackle these core use cases.

As a result we've done the following:
 - moved the assignment functionality into a new column at desktop breakpoints for easy at a glance listing of current assignments
 - styled the assign button to be consistent with the site-wide styles (this might need a litte tweak for spacing etc)
 - Changed the 'assigned' state to a link to the edit page for reassignment when necessary
 - Added a progressive enhancement to the submit form to assign via AJAX request when available, for easy rapid assignments. This will gracefully fallback to a regular HTTP form submit with a flash confirmation message when javasript isn't available. I've also added a minimal reusable 'waiting' spinner to display while the request is in flight (again, some styling might be necessary for this).

Squashed commits for reference:

 - First attempt at backend logic for unassigning judgments
 - Refactor out separate /unassign endpoint.
 - Broad-strokes table layout change for 'assign to me' FE design
 - Get rid of unassignments, add link to edit form for reassigning
 - Allow users to be assigned with an AJAX request when available.
 - Style the assignment button

## Trello card / Rollbar error (etc)

https://trello.com/c/ZJLPy4Jn/222-eui-fe-for-assign-to-me-switch

#### Desktop
Screenshot 2022-12-15 at 17 03 10

#### Mobile
Screenshot 2022-12-15 at 17 02 56

#### Desktop
Screenshot 2022-12-15 at 17 01 46

#### Mobile

Screenshot 2022-12-15 at 17 02 08

### CODE REVIEW

File: ds_caselaw_editor_ui/static/js/src/app.js [hunk 1]
```diff
@@ -1,49 +1,80 @@
...
+      url: action,
+      data: form.serialize(),
+      success: function (data) {
+        const assigned_to = data["assigned_to"];
+        loading.replaceWith(
+          "<a aria-busy='false' href='/edit?judgment_uri=" +
+            uri +
+            "#assigned_to'>" +
+            assigned_to +
+            "</a>"
```
[the remark this answers is not in the document]
Comment (dragon-dxw) [reply ds_caselaw_editor_ui/static/js/src/app.js:75]: Worth taking a look at -- the strings should be safe but it'd be nice to have this right.

**Review by dragon-dxw — APPROVED** [review]
Worth taking a look at the automatically highlighted issue, but otherwise great. (The second one wasn't really in your code)

File: ds_caselaw_editor_ui/static/js/src/app.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (timcowlishaw) [reply ds_caselaw_editor_ui/static/js/src/app.js]: have added an encodeURIComponent :-)

File: judgments/views.py [hunk 2]
```diff
@@ -267,8 +267,14 @@
...
-    messages.success(request, "Judgment assigned to you.")
-    return redirect(target_uri)
+    if request.headers.get("X-Requested-With") == "XMLHttpRequest":
+        return HttpResponse(
+            json.dumps({"assigned_to": request.user.username}),
+            content_type="application/json",
+        )
+    else:
+        messages.success(request, "Judgment assigned to you.")
+        return redirect(target_uri)
```
[the remark this answers is not in the document]
Comment (timcowlishaw) [reply judgments/views.py:277]: @jacksonj04 @dragon-dxw  I suspect we can get rid of this by checking that the supplied URL is local, but i'm afraid that's beyond the limits of my django currently. I suspect github will remind us again in the future though so will ignore for now.

**Pull request merged**